skunkworks_crow/src/main/java/org/odk/share/activities/MainActivity.java

```diff
@@ -237,4 +236,42 @@ private void addListItemDivider() {
 
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        switch (requestCode) {
+            case STORAGE_PERMISSION_REQUEST_CODE:
+                if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    setUpLoader();
+                } else {
+                    showNoPermissionView();
```

Comment:
The app should be closed instead of making the empty view as the trigger to re-initialize the permission request

Reply:
made changes to close the app if the permission is denied